Orange/widgets/utils/scaling.py

```diff
@@ -30,53 +32,58 @@ def rescale_data(self):
 
     def _compute_domain_data_stat(self):
         stt = self.domain_data_stat = \
-            getCached(self.data, DomainBasicStats, (self.data,))
-        for index in range(len(self.domain)):
-            attr = self.domain[index]
+            getCached(self.data, DomainBasicStats, (self.data, True))
+        domain = self.domain
+        for attr in chain(domain.variables, domain.metas):
             if attr.is_discrete:
                 self.attr_values[attr] = [0, len(attr.values)]
             elif attr.is_continuous:
-                self.attr_values[attr] = [stt[index].min, stt[index].max]
+                self.attr_values[attr] = [stt[attr].min, stt[attr].max]
 
     def _compute_scaled_data(self):
         data = self.data
         # We cache scaled_data and validArray to share them between widgets
         cached = getCached(data, "visualizationData")
         if cached:
-            self.original_data, self.scaled_data, self.valid_data_array = cached
+            self.data, self.scaled_data, self.valid_data_array = cached
             return
 
         Y = data.Y if data.Y.ndim == 2 else np.atleast_2d(data.Y).T
-        self.original_data = np.hstack((data.X, Y)).T
-        self.scaled_data = no_jit = self.original_data.copy()
-        self.valid_data_array = np.isfinite(no_jit)
-        for index in range(len(data.domain)):
-            attr = data.domain[index]
+        if np.any(data.metas):
+            all_data = (data.X, Y, data.metas)
+        else:
+            all_data = (data.X, Y)
+        all_data = np.hstack(all_data).T
```

Comment:
This can have dtype==object when stacking with metas which will cause np.isfinite to fail 2 lines below

Reply:
Thanks. Fixed.

Reply:
Fixed here, but I get similar errors elsewhere (see sentry report).
Try fixing all metas dtype problems (converting all primitive metas to float might be enough, maybe something else is also needed). Then please test thoroughly! Use at least the workflow from the sentry report.

Reply:
I found another error caused by Corpus (issue: https://sentry.io/biolab/orange3-text-bl/issues/399242697/ )